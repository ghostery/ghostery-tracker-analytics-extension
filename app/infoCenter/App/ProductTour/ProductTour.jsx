import React, { Component } from 'react';
import PropTypes from 'prop-types';
import ClassNames from 'classnames';
import Collapse from 'react-bootstrap/Collapse';
import OverlayTrigger from 'react-bootstrap/OverlayTrigger';
import Tooltip from 'react-bootstrap/Tooltip';
import './ProductTour.scss';

class ProductTour extends Component {
  constructor(props) {
    super(props);

    this.state = {
      currentView: 0,
      featureDescriptions: [
        { view: 1, title: 'Tracker Request List', text: 'This list displays all trackers and scripts operating on a website. It includes size and latency data, and can be expanded to show request URLs. “Favorite” a tracker to pin it to the top of your list whenever it appears on a page. You can manage favorites in your Settings panel.' },
        { view: 2, title: 'Tracker Timeline', text: 'The Timeline visualizes each tracker’s ping behavior in real-time as a page loads. Hover over data points in an expanded visual to view script size and latency information. From the header, select which page events to show, and toggle between a linear or logarithmic display.' },
        { view: 3, title: 'Tracker Distribution Graph', text: 'The Tracker Distribution Chart displays trackers by size and latency in a scatterplot, making it easy to identify outliers at a glance. Hover over a data point to see details, and click to view corresponding tracker data in the Timeline.' },
        { view: 4, title: 'Page Latency Graph', text: 'The Page Latency graph breaks down the initial load of a web page into categories of processes organized by events, such as the time spent making a request to a server and receiving a response.' },
        { view: 5, title: 'Page Size Graph', text: 'The Page Size panel shows how much of your local memory is being allocated for certain types of data (eg. images, scripts, fonts, etc.). This graph is updated in real time as you interact with any web page.' },
        { view: 6, title: 'Live/Freeze Mode', text: 'In Live mode, data is streaming to the dashboard in real-time. You can set the toggle to Freeze to pause data streaming and capture a snapshot of tracker activity at that point in time. Toggle back to Live to resume.' },
        { view: 7, title: 'Global Trends', text: 'The Global Trends tab lists privacy-related statistics and trends over time for trackers and websites documented in our web-profiling database. Click a tracker to expand it, and hover over the line graphs to view historical stats. You can arrange trackers by favorites, prevalence, or category using the sort options at the top of the list. ' },
      ],
      images: [
        { view: 0, type: 'static', imagePath: '/dist/images/info-center/welcome-to-insights.png' },
        { view: 1, type: 'static', imagePath: '/dist/images/info-center/tracker-request-list.png' },
        {
          view: 2,
          type: 'pulsar',
          imagePath: '/dist/images/info-center/tracker-timeline.png',
          tooltips: [
            'Toggle between linear or logarithmic scaling',
            'Select page events you want displayed on the timelie alongside tracker request pings',
          ],
        },
        {
          view: 3,
          type: 'pulsar',
          imagePath: '/dist/images/info-center/tracker-distribution-graph.png',
          tooltips: [
            'Hover over or click a data point to view size, latency, and timeline information',
          ],
        },
        { view: 4, type: 'static', imagePath: '/dist/images/info-center/page-latency-graph.png' },
        { view: 5, type: 'static', imagePath: '/dist/images/info-center/page-size-graph.png' },
        { view: 6, type: 'static', imagePath: '/dist/images/info-center/live-freeze-mode.png' },
        {
          view: 7,
          type: 'pulsar',
          imagePath: '/dist/images/info-center/global-trends.png',
          tooltips: [
            'Sort the trackers by prevalence, category, or favorites for convenient viewing',
            'Web traffic trends over time are visualized by line graphs — hover to see statistics for a specific date',
            'Expand a tracker to see granular information, such as associated websites and tracking methods used',
          ],
        },
      ],
      addPulsars: false,
    };
  }

  componentDidMount() {
    this.setState({ addPulsars: true });
  }

  switchView = (view) => {
    const { toggleBackgroundCurve } = this.props;
    const { currentView } = this.state;

    if (view === currentView) {
      toggleBackgroundCurve(true, false);
      this.setState({ currentView: 0 });
      return;
    }

    if (currentView === 0) { toggleBackgroundCurve(false, true); }
    this.setState({ currentView: view });
  }

  renderFeatureDescriptions = () => {
    const { featureDescriptions, currentView } = this.state;
    const caretClasses = featureDescriptions.map(feature => (
      ClassNames('ProductTour__caret', { flipped: feature.view === currentView })
    ));

    return featureDescriptions.map(feature => (
      <div className="ProductTour__feature" key={`description-${feature.view}`}>
        <div
          className="ProductTour__featureHeader d-flex align-items-center justify-content-between"
          onClick={() => this.switchView(feature.view)}
        >
          <h2>{feature.title}</h2>
          <img
            className={caretClasses[feature.view - 1]}
            alt={`Expand or Collapse ${feature.title}`}
            src="/dist/images/app/caret-down.svg"
          />
        </div>
        <Collapse in={feature.view === currentView}>
          <div className="ProductTour__featureDescription">
            <div><p>{feature.text}</p></div>
          </div>
        </Collapse>
      </div>
    ));
  }

  renderPulsars = (image) => {
    const digitsToWords = ['Zero', 'One', 'Two', 'Three', 'Four', 'Five', 'Six', 'Seven'];
    const pulsarKeyAndClass = (view, index) => (
      `image${digitsToWords[view]}Pulsar${digitsToWords[index]}`
    );

    return image.tooltips.map((tooltip, i) => (
      <OverlayTrigger
        key={pulsarKeyAndClass(image.view, i)}
        placement="auto"
        overlay={(
          <Tooltip className="ProductTour__tooltipPurpleBorder tooltipLeft">
            {tooltip}
          </Tooltip>
        )}
      >
        <div className={`ProductTour__pulsar ${pulsarKeyAndClass(image.view, i)} d-flex justify-content-center align-items-center`}>
          <div />
        </div>
      </OverlayTrigger>
    ));
  }

  renderImages = () => {
    const { currentView, images, addPulsars } = this.state;
    const imageClasses = images.map(image => (
      ClassNames('ProductTour__imageDisplay', `image-${image.view}`, {
        show: image.view === currentView,
        liveFreezeSize: currentView === 6,
      })
    ));

    return images.map(image => (
      <div
        className={imageClasses[image.view]}
        key={`image-${image.view}`}
      >
        <img
          className="ProductTour__image"
          alt={`Feature ${currentView}`}
          src={image.imagePath}
        />
        <div className="ProductTour__pulsarContainer">
          {addPulsars && image.type === 'pulsar' && this.renderPulsars(image)}
        </div>
      </div>
    ));
  }

  render() {
    const { currentView } = this.state;
    const imageContainerClasses = ClassNames(
      'ProductTour__imageContainer',
      'd-flex',
      'align-items-center',
      'justify-content-center',
      { addMarginTop: currentView !== 0 },
    );

    return (
      <div className="ProductTour">
        <div className="d-flex justify-content-between">
          <div className="ProductTour__featureContainer">
            <h2>Welcome to Ghostery Insights</h2>
            <p>
              Insights offers comprehensive, real-time analysis into tracker activity on a page. Learn about the dashboard and features below.
            </p>
            {this.renderFeatureDescriptions()}
          </div>
          <div className={imageContainerClasses}>
            {this.renderImages()}
          </div>
        </div>
      </div>
    );
  }
}

ProductTour.propTypes = {
  toggleBackgroundCurve: PropTypes.func.isRequired,
};

export default ProductTour;
