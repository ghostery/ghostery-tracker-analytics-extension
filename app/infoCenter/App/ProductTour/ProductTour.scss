@import '../../../utils/scss/color-variables.scss';

.ProductTour {
  margin: 30px 0 100px;
}

.ProductTour__featureContainer > p {
  margin-bottom: 70px;
  width: 400px;
  line-height: 30px;
}

.ProductTour__feature {
  margin-bottom: 15px;
  width: 350px;
}

.ProductTour__featureHeader {
  background-color: $opaque-background-medium;
  padding: 9px 12px;
  cursor: pointer;

  h2 {
    margin: 0;
  }
}

.ProductTour__caret {
  width: 9px;
  height: 5px;

  &.flipped {
    transform: rotate(180deg);
  }
}

.ProductTour__featureDescription {
  background-color: $opaque-background-white-lighter;

  > div {
    padding: 1px 20px;
  }

  > div > p {
    line-height: 23px;
  }
}

.ProductTour__imageContainer {
  margin-top: 20px;
  width: 450px;
  height: 550px;
  transition: margin-top .3s;

  &.addMarginTop {
    margin-top: 210px;
  }
}

.ProductTour__imageDisplay {
  position: absolute;
  opacity: 0;
  width: 450px;
  height: 550px;
  pointer-events: none;
  transition: opacity .3s;

  &.show {
    opacity: 1;
    pointer-events: auto;
  }

  > img {
    object-fit: contain;
    width: 450px;
    height: 550px;
    transition: height .3s;
  }

  &.liveFreezeSize > img {
    height: 513px;
  }
}

.ProductTour__pulsarContainer {
  position: relative;
}

.ProductTour__pulsar {
  position: absolute;
  background-color: $opaque-pulsar-purple;
  border-radius: 20px;
  width: 10px;
  height: 10px;
  cursor: pointer;
  animation: borderPulse 2s ease infinite;

  &.imageTwoPulsarZero {
    top: -236px;
    left: 202px;
  }

  &.imageTwoPulsarOne {
    top: -236px;
    left: 375px;
  }

  &.imageThreePulsarZero {
    top: -366px;
    left: 126px;
  }

  &.imageSevenPulsarZero {
    top: -395px;
    left: 97px;
  }

  &.imageSevenPulsarOne {
    top: -319px;
    left: 91px;
  }

  &.imageSevenPulsarTwo {
    top: -361px;
    left: 148px;
  }

  > div {
    background-color: $pulsar-purple;
    border-radius: 20px;
    width: 10px;
    height: 10px;
  }
}

@keyframes borderPulse {
  75% {
    transform: translate(0, 0);
    width: 10px;
    height: 10px;
  }

  87% {
    transform: translate(-5px, -5px);
    width: 20px;
    height: 20px;
  }

  100% {
    transform: translate(0, 0);
    width: 10px;
    height: 10px;
  }
}

.ProductTour--marginTop {
  margin-top: 200px;
}
