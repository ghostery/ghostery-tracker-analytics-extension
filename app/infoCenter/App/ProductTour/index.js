import ProductTour from './ProductTour.jsx';

export default ProductTour;
