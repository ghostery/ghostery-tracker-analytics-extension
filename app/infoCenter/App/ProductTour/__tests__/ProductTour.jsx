import React from 'react';
import { shallow } from 'enzyme';

import ProductTour from '../ProductTour';

describe('app/infoCenter/ProductTour', () => {
  describe('Snapshot tests with react-test-renderer', () => {
    test('ProductTour component should render', () => {
      const component = shallow(<ProductTour toggleBackgroundCurve={() => {}} />);

      expect(component).toMatchSnapshot();
    });
  });
});
