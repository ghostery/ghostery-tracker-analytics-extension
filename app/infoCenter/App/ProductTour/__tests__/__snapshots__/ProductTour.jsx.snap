// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/infoCenter/ProductTour Snapshot tests with react-test-renderer ProductTour component should render 1`] = `
<div
  className="ProductTour"
>
  <div
    className="d-flex justify-content-between"
  >
    <div
      className="ProductTour__featureContainer"
    >
      <h2>
        Welcome to Ghostery Insights
      </h2>
      <p>
        Insights offers comprehensive, real-time analysis into tracker activity on a page. Learn about the dashboard and features below.
      </p>
      <div
        className="ProductTour__feature"
        key="description-1"
      >
        <div
          className="ProductTour__featureHeader d-flex align-items-center justify-content-between"
          onClick={[Function]}
        >
          <h2>
            Tracker Request List
          </h2>
          <img
            alt="Expand or Collapse Tracker Request List"
            className="ProductTour__caret"
            src="/dist/images/app/caret-down.svg"
          />
        </div>
        <Collapse
          appear={false}
          dimension="height"
          getDimensionValue={[Function]}
          in={false}
          mountOnEnter={false}
          timeout={300}
          unmountOnExit={false}
        >
          <div
            className="ProductTour__featureDescription"
          >
            <div>
              <p>
                This list displays all trackers and scripts operating on a website. It includes size and latency data, and can be expanded to show request URLs. “Favorite” a tracker to pin it to the top of your list whenever it appears on a page. You can manage favorites in your Settings panel.
              </p>
            </div>
          </div>
        </Collapse>
      </div>
      <div
        className="ProductTour__feature"
        key="description-2"
      >
        <div
          className="ProductTour__featureHeader d-flex align-items-center justify-content-between"
          onClick={[Function]}
        >
          <h2>
            Tracker Timeline
          </h2>
          <img
            alt="Expand or Collapse Tracker Timeline"
            className="ProductTour__caret"
            src="/dist/images/app/caret-down.svg"
          />
        </div>
        <Collapse
          appear={false}
          dimension="height"
          getDimensionValue={[Function]}
          in={false}
          mountOnEnter={false}
          timeout={300}
          unmountOnExit={false}
        >
          <div
            className="ProductTour__featureDescription"
          >
            <div>
              <p>
                The Timeline visualizes each tracker’s ping behavior in real-time as a page loads. Hover over data points in an expanded visual to view script size and latency information. From the header, select which page events to show, and toggle between a linear or logarithmic display.
              </p>
            </div>
          </div>
        </Collapse>
      </div>
      <div
        className="ProductTour__feature"
        key="description-3"
      >
        <div
          className="ProductTour__featureHeader d-flex align-items-center justify-content-between"
          onClick={[Function]}
        >
          <h2>
            Tracker Distribution Graph
          </h2>
          <img
            alt="Expand or Collapse Tracker Distribution Graph"
            className="ProductTour__caret"
            src="/dist/images/app/caret-down.svg"
          />
        </div>
        <Collapse
          appear={false}
          dimension="height"
          getDimensionValue={[Function]}
          in={false}
          mountOnEnter={false}
          timeout={300}
          unmountOnExit={false}
        >
          <div
            className="ProductTour__featureDescription"
          >
            <div>
              <p>
                The Tracker Distribution Chart displays trackers by size and latency in a scatterplot, making it easy to identify outliers at a glance. Hover over a data point to see details, and click to view corresponding tracker data in the Timeline.
              </p>
            </div>
          </div>
        </Collapse>
      </div>
      <div
        className="ProductTour__feature"
        key="description-4"
      >
        <div
          className="ProductTour__featureHeader d-flex align-items-center justify-content-between"
          onClick={[Function]}
        >
          <h2>
            Page Latency Graph
          </h2>
          <img
            alt="Expand or Collapse Page Latency Graph"
            className="ProductTour__caret"
            src="/dist/images/app/caret-down.svg"
          />
        </div>
        <Collapse
          appear={false}
          dimension="height"
          getDimensionValue={[Function]}
          in={false}
          mountOnEnter={false}
          timeout={300}
          unmountOnExit={false}
        >
          <div
            className="ProductTour__featureDescription"
          >
            <div>
              <p>
                The Page Latency graph breaks down the initial load of a web page into categories of processes organized by events, such as the time spent making a request to a server and receiving a response.
              </p>
            </div>
          </div>
        </Collapse>
      </div>
      <div
        className="ProductTour__feature"
        key="description-5"
      >
        <div
          className="ProductTour__featureHeader d-flex align-items-center justify-content-between"
          onClick={[Function]}
        >
          <h2>
            Page Size Graph
          </h2>
          <img
            alt="Expand or Collapse Page Size Graph"
            className="ProductTour__caret"
            src="/dist/images/app/caret-down.svg"
          />
        </div>
        <Collapse
          appear={false}
          dimension="height"
          getDimensionValue={[Function]}
          in={false}
          mountOnEnter={false}
          timeout={300}
          unmountOnExit={false}
        >
          <div
            className="ProductTour__featureDescription"
          >
            <div>
              <p>
                The Page Size panel shows how much of your local memory is being allocated for certain types of data (eg. images, scripts, fonts, etc.). This graph is updated in real time as you interact with any web page.
              </p>
            </div>
          </div>
        </Collapse>
      </div>
      <div
        className="ProductTour__feature"
        key="description-6"
      >
        <div
          className="ProductTour__featureHeader d-flex align-items-center justify-content-between"
          onClick={[Function]}
        >
          <h2>
            Live/Freeze Mode
          </h2>
          <img
            alt="Expand or Collapse Live/Freeze Mode"
            className="ProductTour__caret"
            src="/dist/images/app/caret-down.svg"
          />
        </div>
        <Collapse
          appear={false}
          dimension="height"
          getDimensionValue={[Function]}
          in={false}
          mountOnEnter={false}
          timeout={300}
          unmountOnExit={false}
        >
          <div
            className="ProductTour__featureDescription"
          >
            <div>
              <p>
                In Live mode, data is streaming to the dashboard in real-time. You can set the toggle to Freeze to pause data streaming and capture a snapshot of tracker activity at that point in time. Toggle back to Live to resume.
              </p>
            </div>
          </div>
        </Collapse>
      </div>
      <div
        className="ProductTour__feature"
        key="description-7"
      >
        <div
          className="ProductTour__featureHeader d-flex align-items-center justify-content-between"
          onClick={[Function]}
        >
          <h2>
            Global Trends
          </h2>
          <img
            alt="Expand or Collapse Global Trends"
            className="ProductTour__caret"
            src="/dist/images/app/caret-down.svg"
          />
        </div>
        <Collapse
          appear={false}
          dimension="height"
          getDimensionValue={[Function]}
          in={false}
          mountOnEnter={false}
          timeout={300}
          unmountOnExit={false}
        >
          <div
            className="ProductTour__featureDescription"
          >
            <div>
              <p>
                The Global Trends tab lists privacy-related statistics and trends over time for trackers and websites documented in our web-profiling database. Click a tracker to expand it, and hover over the line graphs to view historical stats. You can arrange trackers by favorites, prevalence, or category using the sort options at the top of the list. 
              </p>
            </div>
          </div>
        </Collapse>
      </div>
    </div>
    <div
      className="ProductTour__imageContainer d-flex align-items-center justify-content-center"
    >
      <div
        className="ProductTour__imageDisplay image-0 show"
        key="image-0"
      >
        <img
          alt="Feature 0"
          className="ProductTour__image"
          src="/dist/images/info-center/welcome-to-insights.png"
        />
        <div
          className="ProductTour__pulsarContainer"
        />
      </div>
      <div
        className="ProductTour__imageDisplay image-1"
        key="image-1"
      >
        <img
          alt="Feature 0"
          className="ProductTour__image"
          src="/dist/images/info-center/tracker-request-list.png"
        />
        <div
          className="ProductTour__pulsarContainer"
        />
      </div>
      <div
        className="ProductTour__imageDisplay image-2"
        key="image-2"
      >
        <img
          alt="Feature 0"
          className="ProductTour__image"
          src="/dist/images/info-center/tracker-timeline.png"
        />
        <div
          className="ProductTour__pulsarContainer"
        >
          <OverlayTrigger
            defaultOverlayShown={false}
            key="imageTwoPulsarZero"
            overlay={
              <Tooltip
                className="ProductTour__tooltipPurpleBorder tooltipLeft"
                placement="right"
              >
                Toggle between linear or logarithmic scaling
              </Tooltip>
            }
            placement="auto"
            trigger={
              Array [
                "hover",
                "focus",
              ]
            }
          >
            <div
              className="ProductTour__pulsar imageTwoPulsarZero d-flex justify-content-center align-items-center"
            >
              <div />
            </div>
          </OverlayTrigger>
          <OverlayTrigger
            defaultOverlayShown={false}
            key="imageTwoPulsarOne"
            overlay={
              <Tooltip
                className="ProductTour__tooltipPurpleBorder tooltipLeft"
                placement="right"
              >
                Select page events you want displayed on the timelie alongside tracker request pings
              </Tooltip>
            }
            placement="auto"
            trigger={
              Array [
                "hover",
                "focus",
              ]
            }
          >
            <div
              className="ProductTour__pulsar imageTwoPulsarOne d-flex justify-content-center align-items-center"
            >
              <div />
            </div>
          </OverlayTrigger>
        </div>
      </div>
      <div
        className="ProductTour__imageDisplay image-3"
        key="image-3"
      >
        <img
          alt="Feature 0"
          className="ProductTour__image"
          src="/dist/images/info-center/tracker-distribution-graph.png"
        />
        <div
          className="ProductTour__pulsarContainer"
        >
          <OverlayTrigger
            defaultOverlayShown={false}
            key="imageThreePulsarZero"
            overlay={
              <Tooltip
                className="ProductTour__tooltipPurpleBorder tooltipLeft"
                placement="right"
              >
                Hover over or click a data point to view size, latency, and timeline information
              </Tooltip>
            }
            placement="auto"
            trigger={
              Array [
                "hover",
                "focus",
              ]
            }
          >
            <div
              className="ProductTour__pulsar imageThreePulsarZero d-flex justify-content-center align-items-center"
            >
              <div />
            </div>
          </OverlayTrigger>
        </div>
      </div>
      <div
        className="ProductTour__imageDisplay image-4"
        key="image-4"
      >
        <img
          alt="Feature 0"
          className="ProductTour__image"
          src="/dist/images/info-center/page-latency-graph.png"
        />
        <div
          className="ProductTour__pulsarContainer"
        />
      </div>
      <div
        className="ProductTour__imageDisplay image-5"
        key="image-5"
      >
        <img
          alt="Feature 0"
          className="ProductTour__image"
          src="/dist/images/info-center/page-size-graph.png"
        />
        <div
          className="ProductTour__pulsarContainer"
        />
      </div>
      <div
        className="ProductTour__imageDisplay image-6"
        key="image-6"
      >
        <img
          alt="Feature 0"
          className="ProductTour__image"
          src="/dist/images/info-center/live-freeze-mode.png"
        />
        <div
          className="ProductTour__pulsarContainer"
        />
      </div>
      <div
        className="ProductTour__imageDisplay image-7"
        key="image-7"
      >
        <img
          alt="Feature 0"
          className="ProductTour__image"
          src="/dist/images/info-center/global-trends.png"
        />
        <div
          className="ProductTour__pulsarContainer"
        >
          <OverlayTrigger
            defaultOverlayShown={false}
            key="imageSevenPulsarZero"
            overlay={
              <Tooltip
                className="ProductTour__tooltipPurpleBorder tooltipLeft"
                placement="right"
              >
                Sort the trackers by prevalence, category, or favorites for convenient viewing
              </Tooltip>
            }
            placement="auto"
            trigger={
              Array [
                "hover",
                "focus",
              ]
            }
          >
            <div
              className="ProductTour__pulsar imageSevenPulsarZero d-flex justify-content-center align-items-center"
            >
              <div />
            </div>
          </OverlayTrigger>
          <OverlayTrigger
            defaultOverlayShown={false}
            key="imageSevenPulsarOne"
            overlay={
              <Tooltip
                className="ProductTour__tooltipPurpleBorder tooltipLeft"
                placement="right"
              >
                Web traffic trends over time are visualized by line graphs — hover to see statistics for a specific date
              </Tooltip>
            }
            placement="auto"
            trigger={
              Array [
                "hover",
                "focus",
              ]
            }
          >
            <div
              className="ProductTour__pulsar imageSevenPulsarOne d-flex justify-content-center align-items-center"
            >
              <div />
            </div>
          </OverlayTrigger>
          <OverlayTrigger
            defaultOverlayShown={false}
            key="imageSevenPulsarTwo"
            overlay={
              <Tooltip
                className="ProductTour__tooltipPurpleBorder tooltipLeft"
                placement="right"
              >
                Expand a tracker to see granular information, such as associated websites and tracking methods used
              </Tooltip>
            }
            placement="auto"
            trigger={
              Array [
                "hover",
                "focus",
              ]
            }
          >
            <div
              className="ProductTour__pulsar imageSevenPulsarTwo d-flex justify-content-center align-items-center"
            >
              <div />
            </div>
          </OverlayTrigger>
        </div>
      </div>
    </div>
  </div>
</div>
`;
